org-formation/040-budgets/_tasks.yaml

```diff
@@ -13,6 +13,16 @@ BudgetAlarms:
     budgetAmount: !GetAtt CurrentAccount.Tags.budget-alarm-threshold
     emailAddress: !GetAtt CurrentAccount.Tags.budget-alarm-threshold-email-recipient
 
+CostReportLambda:
+  Type: update-stacks
+  Template: !Sub 'https://${AdminCentralCfnBucket}.s3.amazonaws.com/lambda-cost-reports/0.0.3/lambda-cost-reports.yaml'
+  StackName: !Sub '${resourcePrefix}-cost-reports-lambda'
+  DefaultOrganizationBinding:
+    Account: !Ref MasterAccount
+    Region: !Ref primaryRegion
+  Parameters:
+    bucketName: !Sub "${resourcePrefix}-cost-reports"
```

Comment:
hmm, lambda depends on the bucket and the bucket depends on the lambda in line 36.  circular dependency?

Reply:
almost, by putting the event configuration on the bucket resource rather than the lambda resource, we make the bucket dependent on the lambda; and the lambda also depends on a managed policy that is dependent on the name of the bucket. but by separating the bucket from the bucket name, we can make both stacks dependent on the bucket name without having a circular dependency between the lambda, lambda policy, and bucket